Add Sokoban Puzzles

Miserlou opened the pull request:
Ex:


```
This is a sokoban puzzle. Please solve it. Your solution must be a string of characters, ex: LDURRUDL

+ + + + + + +
+ * - @ - X +
+ + - @ - + +
+ X - - - $ +
+ + + + + + +

* - The player
% - The player on a goal
@ - A box
X - A goal
$ - A box on a goal
+ - A wall
- - An empty position
```

Configurable sizes and number of boxes for different difficulty levels.

DS-R1 and CGPT-O1 can solve these, but it takes R1 a very long time for the simplest case (five minutes ish.) O1 does it quickly.

Hacked up from [this project](https://github.com/xbandrade/sokoban-solver-generator) (MIT). There's still some stuff in there that can be stripped out (particularly the GUI/sprite stuff), but it's all quite intertwined.

andreaskoepf commented on reasoning_gym/games/sokoban.py:
```diff
@@ -0,0 +1,120 @@
+
+        grid_list = [list(line) for line in entry["metadata"]["gamestr"].replace(" ", "").strip().split("\n")]
+        matrix = np.array(grid_list)
+        state = get_state(matrix)
```
do we catch wrong/malformed inputs, should there be an try..except around the parsing?

andreaskoepf commented on reasoning_gym/games/sokoban.py:
```diff
@@ -0,0 +1,120 @@
+        # rng = Random(self.seed + idx)
+
+        # Make the Sokoban!
+        (game, matrix, gamestr) = generate(seed=self.seed + idx + 1)
```
Any special reason for `+1` here

andreaskoepf commented on reasoning_gym/games/sokoban.py:
```diff
@@ -0,0 +1,120 @@
+                - answer: str, a solution string
+                - metadata: dict with generation parameters
+        """
+        # rng = Random(self.seed + idx)
```
could be removed probably

andreaskoepf commented on pyproject.toml:
```diff
@@ -18,6 +18,7 @@ dependencies = [
   "magiccube==0.3.0",
   "pycosat==0.6.6",
   "pyfiglet==1.0.2",
+  "pygame==2.6.1",
```
pygame is a larger dependency. Do we have any chance to get rid of it?

andreaskoepf commented:
Please add the sokuban solver in the #56 NOTICE.txt issue.

Miserlou commented:
Addressed comments. Pygame was more integrated into this project than I expected and made hacking it up a bit annoying. If I were to do it again, I'd probably just ask the robot to write it from scratch, though this library did come with a few solving strategies to play with. It's only a 19M dep, not toooo bad at least.

andreaskoepf commented:
replaced by #77
